client/my-sites/plans/current-plan/purchases-listing.jsx

```diff
@@ -97,49 +93,35 @@ class PurchasesListing extends Component {
 
 	getProductPurchases() {
 		return (
-			filter(
-				this.props.purchases,
-				( purchase ) =>
-					purchase.active && ( isJetpackProduct( purchase ) || isJetpackSearch( purchase ) )
-			) ?? null
+			this.props.purchases?.filter(
+				( purchase ) => purchase.active && isJetpackProduct( purchase )
+			) ?? []
 		);
 	}
 
 	getTitle( purchase ) {
-		const { currentPlanSlug } = this.props;
-
-		if ( isJetpackSearch( purchase.productSlug ) ) {
-			return getJetpackProductDisplayName( purchase );
-		}
+		const { currentPlan } = this.props;
 
 		if ( isJetpackProduct( purchase ) ) {
 			return getJetpackProductDisplayName( purchase );
 		}
 
-		if ( currentPlanSlug ) {
-			const planObject = getPlan( currentPlanSlug );
+		if ( currentPlan ) {
+			const planObject = getPlan( currentPlan.productSlug );
 			return planObject.getTitle();
 		}
 
 		return null;
 	}
 
-	getTagline( purchase ) {
-		const { currentPlanSlug, translate } = this.props;
-
-		if ( isJetpackProduct( purchase ) ) {
-			return getJetpackProductTagline( purchase );
-		}
-
-		if ( isJetpackSearch( purchase ) ) {
-			return getJetpackProductTagline( 'search' );
-		}
+	getPlanTagline( plan ) {
+		const { translate } = this.props;
 
-		const productPurchases = this.getProductPurchases();
-		if ( currentPlanSlug ) {
-			const planObject = getPlan( currentPlanSlug );
+		if ( plan ) {
+			const productPurchases = this.getProductPurchases().map( ( { productSlug } ) => productSlug );
+			const planObject = getPlan( plan.productSlug );
 			return (
-				planObject.getTagline?.( productPurchases[ 0 ]?.productSlug ) ??
+				planObject.getTagline?.( productPurchases ) ??
```

Comment:
We pass to the `getTagline` function all products the site owns. With this, Jetpack Free can have a context-driven tagline.